vernac/g_vernac.ml4

```diff
@@ -75,27 +75,50 @@ GEXTEND Gram
       | IDENT "Redirect"; s = ne_string; c = located_vernac -> VernacRedirect (s, c)
       | IDENT "Timeout"; n = natural; v = vernac_control -> VernacTimeout(n,v)
       | IDENT "Fail"; v = vernac_control -> VernacFail v
-      | (f, v) = vernac -> VernacExpr(f, v) ]
+      | (f, v) = decorated_vernac -> VernacExpr (f, v) ]
     ]
   ;
+  decorated_vernac:
+  [[ a = attributes ; (f, v) = vernac -> (List.append a f, v)
+  | fv = vernac -> fv
+  ]]
+  ;
+  attributes:
+  [[ "#[" ; a = attribute_list ; "]" -> a
```

Comment:
In camlp5 slang this means that `#[` is now a keyword.
I think CI will tell if `"#" "["` is preferable, I've no clue if people have notations using `#` and `[` separately and use both in the same expression.